Client/Frontend/Browser/Search/BraveSearchManager.swift

```diff
@@ -0,0 +1,280 @@
+        
+        URLSession(configuration: .ephemeral)
+            .dataTaskPublisher(for: request)
+            .tryMap { output -> String in
+                guard let response = output.response as? HTTPURLResponse,
+                      response.statusCode >= 200 && response.statusCode < 300 else {
+                    throw "Invalid response"
+                }
+                
+                guard let stringFromData = String(data: output.data, encoding: .utf8) else {
+                    throw "Failed to decode string from data"
+                }
+                
+                return stringFromData.javaScriptEscapedString
+            }
+            .receive(on: DispatchQueue.main)
+            .sink(receiveCompletion: { status in
+                switch status {
+                case .failure(let error):
+                    log.error("Error: \(error)")
+                case .finished:
+                    break
+                }
+            },
+            receiveValue: { [weak self] data in
+                self?.fallbackQueryResult = data
+                completion(data)
+            })
+            .store(in: &cancellables)
+    }
+}
+
+// MARK: - URLSessionDataDelegate
+// The BraveSearch feature can be hidden behind an authentication system.
+// This code helps passing all auth info to the requests we make.
+extension BraveSearchManager: URLSessionDataDelegate {
+    
+    private func findLoginsForProtectionSpace(profile: Profile, challenge: URLAuthenticationChallenge, completion: @escaping (URLCredential?) -> Void) {
+        profile.logins.getLoginsForProtectionSpace(challenge.protectionSpace) >>== { cursor in
+            guard cursor.count >= 1 else {
+                completion(nil)
+                return
+            }
+
+            let logins = cursor.asArray()
+            var credentials: URLCredential?
+
+            if logins.count > 1 {
+                credentials = (logins.find { login in
+                    (login.protectionSpace.`protocol` == challenge.protectionSpace.`protocol`) && !login.hasMalformedHostname
+                })?.credentials
+            } else if logins.count == 1, logins.first?.protectionSpace.`protocol` != challenge.protectionSpace.`protocol` {
+                credentials = logins.first?.credentials
+            } else {
+                credentials = logins.first?.credentials
+            }
+            
+            completion(credentials)
+        }
+    }
+    
+    func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+        
+        guard challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodHTTPBasic ||
+              challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodHTTPDigest ||
+              challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodNTLM else {
```

Comment:
do we actually use HTTP Digest or NTLM for search?

Reply:
This is Firefoxes code from our regular requests navigation delegate, we just copied it over
cc @Brandon-T